tracks/javascript/exercises/bob/mentoring.md

```diff
@@ -0,0 +1,77 @@
+This exercise explores string testing:
+- `RegExp`
+- `String` index accessor
+
+### Reasonable solutions
+
+This exercise can be approached with regular expressions, using the `String`
+prototype, or a combination of both. Below are consise solutions for either but
+combinations are valid too, as long as they make sense:
+
+- `isSilence`: Has no letters, digits or punctuation (ignoring whitespace).
+- `isShouting`: All letters are uppercase, and there is at least one letter.
+- `isAsking`: Ends with a question mark (ignoring whitespace).
+
+#### Regular expressions
+```javascript
+const isSilence = message => /^\s*$/.test(message)
+const isShouting = message => /^[^A-Za-z]*[A-Z]+(?:[^a-z]*)$/.test(message)
+const isAsking = message => /\?\s*$/.test(message)
+```
+
+#### String prototype
+```javascript
+const isSilence = message => message.trim() === ''
+const isShouting = message => message.toUpperCase() === message && message.toLowerCase() !== message
+const isAsking = message => message.trim()[message.length - 1] === '?'
```

Comment:
These methods are not on `prototype` right?

Reply:
[They](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toUpperCase) [are](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim).